src/test/java/org/broadinstitute/hellbender/utils/pairhmm/VectorPairHMMUnitTest.java

```diff
@@ -23,22 +24,24 @@
 
     private static final String pairHMMTestData = publicTestDir + "pairhmm-testdata.txt";
 
-   @BeforeClass
-    public void initialize() {
-       try {
-           new VectorLoglessPairHMM(null);
-       } catch (final Exception e) {
-           throw new SkipException("AVX PairHMM is not supported on this system or the library is not available");
-       }
-    }
-
+    // Return a list of supported VectorLoglessPairHMM implementations, skip the test if none are supported
     private List<N2MemoryPairHMM> getHMMs() {
-        final PairHMMNativeArguments args = new PairHMMNativeArguments();
-        args.useDoublePrecision = false;
-        args.maxNumberOfThreads = 1;
-        final N2MemoryPairHMM avxPairHMM = new VectorLoglessPairHMM(args);
-        avxPairHMM.doNotUseTristateCorrection();
-        return Collections.singletonList(avxPairHMM);
+        List<N2MemoryPairHMM> list = new ArrayList<>();
```

Comment:
Isn't this going to skip all or none of them?  When this function is called it will either pass and return a list of Hmms or it will throw a skip exception.  I think it needs to be re-written as a dataprovider that provides each of `Implementation.values()`.  Then move the code to construct the hmm given an implementation into the test so that when it runs it generates one test for each implementation and then those tests are run or skipped individually.